lib/kitchen/driver/ec2.rb

```diff
@@ -289,7 +289,8 @@ def default_ami
       def update_username(state)
         # TODO: if the user explicitly specified the transport's default username,
         # do NOT overwrite it!
-        if instance.transport[:username] == instance.transport.class.defaults[:username]
+        if actual_platform &&
+            instance.transport[:username] == instance.transport.class.defaults[:username]
```

Comment:
Neither of those `instance.transport`s are Nil for me (it's `actual_platform.username` that's coming up Nil when using non-"standard" AMI's, causing the error).

With a fresh setup and me not specifying anything, both of them are set to "root".  I can alter `instance.transport[:username]` by setting `transport.username` in my .kitchen.yml, and i think the default value for it - and `instance.transport.class.defaults[:username]` - are probably both coming from TK transport defaults? (ie; test-kitchen-1.13.0/lib/kitchen/transport/ssh.rb, which is "root")

I guess that's why that equality condition can't just check for `nil?` and the TODO still applies... (if i want to keep my TK username as "root" and i'm using the SSH transport, it looks like it'll always be overwritten by the `actual_platform.username` value?)

I didn't want to tinker with the existing username logic too much with just that comment for guidance, and if both of those `instance.transport` variables are getting defaults, i'm not sure how you'd test if the user was overriding one?  Maybe @jkeiser remembers a bit more about it?